freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/EnvironmentAccessChecker.java

```diff
@@ -0,0 +1,82 @@
+package com.sequenceiq.freeipa.service.freeipa.user;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.cloudera.thunderhead.service.authorization.AuthorizationProto;
+import com.sequenceiq.authorization.resource.AuthorizationResource;
+import com.sequenceiq.authorization.resource.ResourceAction;
+import com.sequenceiq.authorization.resource.RightUtils;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.auth.altus.CrnParseException;
+import com.sequenceiq.cloudbreak.auth.altus.GrpcUmsClient;
+import com.sequenceiq.cloudbreak.auth.security.InternalCrnBuilder;
+import com.sequenceiq.freeipa.service.freeipa.user.model.EnvironmentAccessRights;
+
+import io.grpc.Status;
+import io.grpc.StatusRuntimeException;
+
+public class EnvironmentAccessChecker {
+    private static final Logger LOGGER = LoggerFactory.getLogger(EnvironmentAccessChecker.class);
+
+    private static final String IAM_INTERNAL_ACTOR_CRN = new InternalCrnBuilder(Crn.Service.IAM).getInternalCrnForServiceAsString();
```

Comment:
Since this is Freeipa service, then I would create internal CRN for freeipa service (or environment because freeipa service is not a public service afaik), not for IAM service

Reply:
The internal CRN needs to specify IAM.
See https://github.com/hortonworks/cloudbreak/blob/master/auth-connector/src/main/java/com/sequenceiq/cloudbreak/auth/security/InternalCrnBuilder.java#L12-L15
See https://github.infra.cloudera.com/thunderhead/thunderhead/blob/master/services/libs/protocols/src/main/java/com/cloudera/thunderhead/service/common/crn/Crns.java#L712-L724